tools/c7n_azure/c7n_azure/resources/app_gateway.py

```diff
@@ -33,3 +36,75 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup'
         )
         resource_type = 'Microsoft.Network/applicationGateways'
+
+
+@ApplicationGateway.filter_registry.register('waf')
+class ApplicationGatewayWafFilter(Filter):
+    """
+    Filter Application Gateways using WAF rule configuration
+
+    :example:
+
+    Return all the App Gateways which have rule '944240' disabled.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: test-app-gateway
+            resource: azure.application-gateway
+            filters:
+              - type: waf
+                override_rule: 944240
+                state: disabled
+    """
+
+    schema = type_schema(
+        'waf',
+        required=['override_rule', 'state'],
```

Comment:
perhaps just default state to disabled

Reply:
Apologies for the delay. I have added the enhancement you recommended and additional test cases for it.